gorilla-minitwit/internal/db/db_methods.go

```diff
@@ -0,0 +1,276 @@
+		Where("who_id = ?", userID)
+
+	var followerIDs []int
+
+	// Find the IDs from the subquery
+	if err := subQuery.Find(&followerIDs).Error; err != nil {
+		fmt.Println(err.Error())
+		return nil, err
+	}
+
+	// Use the retrieved followerIDs in the main query
+	config.DB.Table("messages").
+		Select("messages.*, users.*").
+		Joins("JOIN users ON messages.author_id = users.user_id").
+		Where("messages.flagged = ? AND (users.user_id = ? OR users.user_id IN (?))", 0, userID, followerIDs).
+		Order("messages.pub_date desc").
+		Find(&messages)
+
+	if config.DB.Error != nil {
+		fmt.Println(config.DB.Error.Error())
+		return nil, config.DB.Error
+	}
+	return messages, nil
+}
+
+// getFollowing fetches up to `limit` users that the user identified by userID is following
+func GetFollowing(userID string, limit int) ([]map[interface{}]interface{}, error) {
+	var users []models.Users
+	config.DB.
+		Select("users.*").
+		Joins("INNER JOIN followers ON users.user_id = followers.whom_id").
+		Where("followers.who_id = ?", userID).
+		Limit(limit).
+		Find(&users)
+
+	if config.DB.Error != nil {
+		fmt.Println(config.DB.Error.Error())
+		return nil, config.DB.Error
+	}
+
+	// Convert []models.Users to []map[interface{}]interface{}
+	var result []map[interface{}]interface{}
+	for _, user := range users {
+		m := map[interface{}]interface{}{
+			"UserID": user.UserID,
+			"Name":   user.Username,
+			"Email":  user.Email,
+			// Add more fields as needed
+		}
+		result = append(result, m)
+	}
+
+	return result, nil
+}
+
+// adds a new message to the database
+func AddMessage(text string, author_id int) error {
+	currentTime := time.Now().UTC()
+
+	fmt.Println("timestamp:", currentTime)
+
+	newMessage := models.Messages{
+		AuthorID: author_id,
+		Text:     text,
+		PubDate:  currentTime, //TODO: ALIGN W. LADS: IS THIS CORRECT?
+		Flagged:  0,
+	}
+
+	config.DB.Create(&newMessage)
+
+	if config.DB.Error != nil {
+		fmt.Println(config.DB.Error.Error())
+		return config.DB.Error
+	}
+
+	return nil
+}
+
+// followUser adds a new follower to the database
+func FollowUser(userID string, profileUserID string) error {
+
+	userIDInt, errz := strconv.Atoi(userID)
+	profileUserIDInt, errx := strconv.Atoi(profileUserID)
+
+	if errz != nil {
+		fmt.Println(errz.Error())
+		return errz
+	} else if errx != nil {
+		fmt.Println(errx.Error())
+		return errx
+	}
+
+	// following relationship already exists
+	var count int64
+	config.DB.Model(&models.Followers{}).Where("who_id = ? AND whom_id = ?", userIDInt, profileUserIDInt).Count(&count)
+	if count > 0 {
+		return nil
+	}
+
+	newFollower := models.Followers{
+		WhoID:  userIDInt,
+		WhomID: profileUserIDInt,
+	}
+
+	config.DB.Create(&newFollower)
+
+	if config.DB.Error != nil {
+		fmt.Println(config.DB.Error.Error())
+		return config.DB.Error
+	}
+
+	return nil
+}
+
+// unfollowUser removes a follower from the database
+func UnfollowUser(userID string, profileUserID string) error {
+	userIDInt, errz := strconv.Atoi(userID)
```

Comment:
Same thing here. Maybe could be a good idea to create a function for this user ID atois since we are using it in more than one place.

Reply:
But that almost create more code/lines, right? It is just a native a one-line native string-method?